src/contrib/cluster/Akka.Cluster.Sharding/ClusterSharding.cs

```diff
@@ -391,6 +523,36 @@ public IActorRef StartProxy(string typeName, string role, IdExtractor idExtracto
         /// Specifies that this entity type is located on cluster nodes with a specific role. 
         /// If the role is not specified all nodes in the cluster are used.
         /// </param>
+        /// <param name="idExtractor">
+        /// Partial function to extract the entity id and the message to send to the  entity from the incoming message, 
+        /// if the partial function does not match the message will  be `unhandled`, i.e.posted as `Unhandled` messages 
+        /// on the event stream
+        /// </param>
+        /// <param name="shardResolver">
+        /// Function to determine the shard id for an incoming message, only messages that passed the `extractEntityId` will be used
+        /// </param>
+        /// <returns>The actor ref of the <see cref="Sharding.ShardRegion"/> that is to be responsible for the shard.</returns>
+        public async Task<IActorRef> StartProxyAsync(string typeName, string role, IdExtractor idExtractor, ShardResolver shardResolver)
+        {
+            var timeout = _system.Settings.CreationTimeout;
+            var settings = ClusterShardingSettings.Create(_system).WithRole(role);
+            var startMsg = new ClusterShardingGuardian.StartProxy(typeName, settings, idExtractor, shardResolver);
+            var started = await _guardian.Value.Ask<ClusterShardingGuardian.Started>(startMsg, timeout);
+            _regions.TryAdd(typeName, started.ShardRegion);
```

Comment:
This doesn't necessarily matters, as this asynchronous async call will return the same actor ref (see [ClusterShardingGuardian handler](https://github.com/akkadotnet/akka.net/blob/dev/src/contrib/cluster/Akka.Cluster.Sharding/ClusterShardingGuardian.cs#L92)).